### PR TITLE
Add a "meta" RFC for our RFC process!

### PR DESCRIPTION
**Pull request by DFurnes**
This pull request introduces the "RFCs" RFC:

> We want to start using RFCs to discuss (and document) architectural decisions.

## Open Questions
- How do folks feel about the template? I wanted to keep it as lightweight as possible, while also giving a clear outline for things that are worth mentioning in a "good" RFC. Anything I missed?
- How do folks feel about the suggested process (see `README.md`)?

### CODE REVIEW

**Comment by mirie** [comment]
I think the template is great! 

In terms of process, I like the simplicity as well. I have two questions:
1. For this step: `Create a pull request and drop a link in #announce-tech so others can review it.` -- so today you put your request in the standup thread. Is that the place to call out a link to the RFC PR, or is the thinking to drop the link outside of that thread? Do we want to tag people (if we are looking for specific opinions) on the PR OR in the slack message (or both)?
2. What's the expectation we want to set for having some eyeballs on the RFC? Are we hoping for review and _initial_ feedback within say, 5 business days? or less time / more time?

File: 001-rfcs.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+# RFCs
+
+We want to start using RFCs to discuss (and document) architectural decisions.
+
+**Discussion:** https://github.com/DoSomething/rfcs/pull/1
+
+## Problem
+
+Ideas tend to get stuck in one person's head by default (or lost in 1:1 conversations). As our team grows, we want to make it easy for everyone to take part in architectural discussions & see where we're thinking about taking our platform.
```
Comment (katiecrane) [inline 001-rfcs.md]: I see this mainly referring to architectural discussions, but I know we've talked about using RFCs for how our team operates and other non-architectural processes. Would it make sense to update the wording here to show that the suggestions don't have to be architectural?

**Comment by DFurnes** [comment]
> For this step: `Create a pull request and drop a link in #announce-tech so others can review it.` -- so today you put your request in the standup thread. Is that the place to call out a link to the RFC PR, or is the thinking to drop the link outside of that thread?

I think just dropping it in the channel (or even adding notifications for new PRs created in this repo) seems fine. I just called out in the standup thread to make sure it didn't get lost at the end of the day!

>  Do we want to tag people (if we are looking for specific opinions) on the PR OR in the slack message (or both)?

Definitely!

> What's the expectation we want to set for having some eyeballs on the RFC? Are we hoping for review and _initial_ feedback within say, 5 business days? or less time / more time?

Hmm, I'm not sure! Let's see how things go and perhaps add some guidelines in the future?

**Comment by DFurnes** [comment]
I've added a "historical" RFC for the [environment naming rules](https://github.com/DoSomething/rfcs/blob/master/002-resource-naming.md) we standardized last year, and another example RFC for continued GraphQL investment in https://github.com/DoSomething/rfcs/pull/3!

File: 001-rfcs.md (hunk 1 above)
Comment (DFurnes) [reply 001-rfcs.md]: Mmmm, good point! I tried re-phrasing this in 8baebe7.

**Review by katiecrane — APPROVED** [review]

File: 001-rfcs.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+
+# RFCs
+
+We want to start using RFCs to discuss (and document) architectural decisions.
+
+**Discussion:** https://github.com/DoSomething/rfcs/pull/1
```
Comment (mendelB) [inline 001-rfcs.md:6]: I really dig the meta-ness of this self-reference, within the meta RFC for the RFC! ♻️

**Review by mendelB — APPROVED** [review]
👏 Yes Yes Yes! This makes great sense!

**Review by weerd — APPROVED** [review]
🎉

**Comment by weerd** [comment]
Also, thoughts on having a directory in the repo that contains all the RFCs instead of top level? Just for organization purposes.

**Comment by DFurnes** [comment]
> Also, thoughts on having a directory in the repo that contains all the RFCs instead of top level? Just for organization purposes.

Since there's only two non-RFC files (that are really just for GitHub to parse), I think I'd rather avoid making people take an extra click to see the list. Can always re-evaluate if this repo gets messier!

**Comment by DFurnes** [comment]
Rebased to fix a merge conflict with the README. 💥

File: 001-rfcs.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Problem
+
+Ideas tend to get stuck in one person's head by default (or lost in 1:1 conversations). As our team grows, we want to make it easy for everyone to take part in architectural/process discussions & see ideas we've committed to trying.
+
+## Proposal
+
+We should adopt a simple RFC process, inspired by the process used by the engineering teams at [Artsy](https://artsy.github.io/blog/2019/04/11/on-an-rfcs-process/) and [GOV.UK](https://github.com/alphagov/govuk-rfcs) and on larger open-source projects like [React](https://github.com/reactjs/rfcs) or [Rust](https://github.com/rust-lang/rfcs). (And while their namesake comes from the [exhaustive IETF standards documents](https://www.rfc-editor.org/rfc-index.html), we definetely don't want to replicate that!)
```
Comment (mshmsh5000) [inline 001-rfcs.md]: _definetely_ ➡️ _definitely_

File: 001-rfcs.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+By documenting architectural goals in a semi-formal manner like this, we hope to increase transparency (since a public document is by definition a lot more transparent than one-one-one discussions). We're also hoping this reduces friction by helping to spread ideas _before_ the pull request review process.
```
Comment (mshmsh5000) [inline 001-rfcs.md]: Small-potatoes, but I'd stick with either _1:1_ or _one-on-one_ across the doc, just so it's easier to scan.

**Review by mshmsh5000 — APPROVED** [review]
Love it! Just two small language comments.

**Comment by rapala61** [comment]
Excited to start using RFCs! Great work 💯 .

**Review by rapala61 — APPROVED** [review]

File: 001-rfcs.md (hunk 3 above)
Comment (DFurnes) [reply 001-rfcs.md]: Ahh! Good catch. 🦅

File: 001-rfcs.md (hunk 4 above)
Comment (DFurnes) [reply 001-rfcs.md]: Agreed! Changed to use "1:1" here as well.

**Comment by DFurnes** [comment]
Thanks everyone for the feedback! I'm going to merge this in! 🚀

**Pull request merged**